raft/node.go

```diff
@@ -42,6 +42,16 @@ type SoftState struct {
 	RaftState StateType
 }
 
+// ReadIndexState provides state for read only query.
+// It's caller's responsibility to send MsgReadIndex first before getting
+// this state from ready, It's also caller's duty to differentiate if this
+// state is what it requests through ReadIndexCtx, eg. given a unique id as
+// ReadIndexCtx
+type ReadIndexState struct {
```

Comment:
how about put this into raft.go and use it as anonymous properties in raft struct, but we still need to define it again in Ready struct as a properties so that it can be read out.